demo/app.py

```diff
@@ -60,8 +66,15 @@ def main():
 
     # Model selection
     st.sidebar.title("Model selection")
-    det_arch = st.sidebar.selectbox("Text detection model", DET_ARCHS)
-    reco_arch = st.sidebar.selectbox("Text recognition model", RECO_ARCHS)
+    framework = st.sidebar.selectbox("Framework", FRAMEWORK)
+    # use framework specific model selection
+    if framework == "PyTorch":
+        os.environ["USE_TORCH"] = "1"
+    else:
+        os.environ["USE_TF"] = "1"
+    (archs_det, archs_reco) = (PT_DET_ARCHS, PT_RECO_ARCHS) if framework == "PyTorch" else (TF_DET_ARCHS, TF_RECO_ARCHS)
+    det_arch = st.sidebar.selectbox("Text detection model", archs_det)
+    reco_arch = st.sidebar.selectbox("Text recognition model", archs_reco)
```

Comment:
I disagree, the backend should be selected via env (i.e. using `USE_TORCH='1' streamlit run demo/app.py` for instance). At runtime otherwise, since we don't have a hold on streamlit internal behaviour, we wouldn't want a user to use one, then try to use the other backend, which might fill up its RAM if the device isn't very beefy